in_toto/model.go

```diff
@@ -0,0 +1,188 @@
+  Inspect []Inspection `json:"inspect"`
+  Keys map[string]Key `json:"keys"`
+  Expires string `json:"expires"`
+  Readme string `json:"readme"`
+}
+
+type Metablock struct {
+  // NOTE: Whenever we want to access an attribute of `Signed` we have to
+  // perform type assertion, e.g. `metablock.Signed.(Layout).Keys`
+  // Maybe there is a better way to store either Layouts or Links in `Signed`?
+  // The notary folks seem to have separate container structs:
+  // https://github.com/theupdateframework/notary/blob/master/tuf/data/root.go#L10-L14
+  // https://github.com/theupdateframework/notary/blob/master/tuf/data/targets.go#L13-L17
+  // I implemented it this way, because there will be several functions that
+  // receive or return a Metablock, where the inner signed has to be inferred
+  // on runtime, e.g. when iterating over links for a layout, and a link can
+  // turn out to be a layout (sublayout)
+  Signed interface{} `json:"signed"`
+  Signatures []Signature `json:"signatures"`
+}
+
+
+func (mb *Metablock) Load(path string) error {
+  // Open file and close before returning
+  jsonFile, err := os.Open(path)
+  defer jsonFile.Close()
+  if err != nil {
+    return err
+  }
+
+  // Read entire file
+  jsonBytes, err := ioutil.ReadAll(jsonFile)
+  if err != nil {
+    return err
+  }
+
+  // Unmarshal JSON into a map of raw messages (signed and signatures)
+  // We can't fully unmarshal immediately, because we need to inspect the
+  // type (link or layout) to decide which data structure to use
+  var rawMb map[string]*json.RawMessage
+  if err := json.Unmarshal(jsonBytes, &rawMb); err != nil {
+    return err
+  }
+
+  // Fully unmarshal signatures part
+  if err := json.Unmarshal(*rawMb["signatures"], &mb.Signatures); err != nil {
+    return err
+  }
+
+  // Temporarily copy signed to opaque map to inspect the `_type` of signed
+  // and create link or layout accordingly
+  var signed map[string]interface{}
+  if err := json.Unmarshal(*rawMb["signed"], &signed); err != nil {
+    return err
+  }
+
+  if signed["_type"] == "link" {
+    var link Link
+    if err := json.Unmarshal(*rawMb["signed"], &link); err != nil {
+    return err
+  }
+    mb.Signed = link
+
+  } else if signed["_type"] == "layout" {
+    var layout Layout
+    if err := json.Unmarshal(*rawMb["signed"], &layout); err != nil {
+    return err
+  }
+    mb.Signed = layout
+
+  } else {
+    return fmt.Errorf(`The '_type' field of the 'signed' part of a metadata
+        file must be one of 'link' or 'layout'`)
+  }
+
+  return nil
+}
+
+func (mb *Metablock) Dump(path string) error {
+  // JSON encode Metablock formatted with newlines and indentation
+  // TODO: parametrize format
+  jsonBytes, err := json.MarshalIndent(mb, "", "  ")
+  if err != nil {
+    return err
+  }
+
+  // Write JSON bytes to the passed path with permissions (-rw-r--r--)
+  err = ioutil.WriteFile(path, jsonBytes, 0644)
+  if err != nil {
+    return err
+  }
+
+  return nil
```

Comment:
Remove the `if` and implicit `else`, just `return err`.

Reply:
The reason why I wrote it that way, is because I thought about (later) enriching the error message with some context information. See https://github.com/in-toto/in-toto-golang/pull/1#discussion_r230003010 for more thoughts on this. I will summarize your comments and my thoughts in a GitHub issue, if that's okay with you.

Reply:
Understood. (I wonder if that's an instance of YAGNI, and/or how you will grep for a puzzling multi-line statement when later wanting to replace it, rather than a plan `return err`.)

Reply:
("Golang-uistic"?)

Reply:
What would be the `value`? `Dump` only has a write side-effect but doesn't return anything (other than an error).